modules/core/src/main/java/org/apache/ignite/internal/util/VersatileReadWriteLock.java

```diff
@@ -364,16 +364,31 @@ private void notifyWriteLockSolicitors() {
         for (Iterator<CompletableFuture<Void>> iterator = writeLockSolicitors.iterator(); iterator.hasNext(); ) {
```

Comment:
Who are `write lock solicitor`? Isn't a solicitor some kind of a lawyer? What does it mean here?

Reply:
Can you please add a javadoc to the `notifyWriteLockSolicitors` method? What is it supposed to do?

Reply:
It seems I chose an incorrect term. 'Solicitor' here is meant to denote the one who waits for a lock to become available for being acquired. I've change 'solicitors' to 'wait set' (it seems to be the term used in the Java spec).

I also added the javadoc to the notifying methods.